### ISSUE TITLE
Unable to debug Xamarin Android App after updating Visual Studio

### ISSUE THREAD

**Issue by ackava** [xamarin/xamarin-android#8075]
### Android application type

Classic Xamarin.Android (MonoAndroid12.0, etc.)

### Affected platform version

VS 2022 Version 17.6.2, Xamarin Android SDK 13.2.0.6

### Description

I was able to run and debug the app, just a simple app, nothing fancy, and I was able to catch exceptions. Now there is some sort of PDB read error.. 

```
Read out of bounds.
System.BadImageFormatException: Read out of bounds.
   at System.Reflection.Throw.OutOfBounds()
   at System.Reflection.Metadata.Ecma335.MethodDebugInformationTableReader.GetDocument(MethodDebugInformationHandle handle)
   at Mono.Debugging.Soft.PortablePdbData.GetDebugInfoFromPdb(MethodMirror method) in D:\a\_work\1\s\External\debugger-libs\Mono.Debugging.Soft\PortablePdbData.cs:line 186
   at Mono.Debugging.Soft.SoftDebuggerBacktrace.CreateStackFrame(StackFrame frame, Int32 frameIndex) in D:\a\_work\1\s\External\debugger-libs\Mono.Debugging.Soft\SoftDebuggerBacktrace.cs:line 157
   at Mono.Debugging.Soft.SoftDebuggerBacktrace.GetStackFrames(Int32 firstIndex, Int32 lastIndex) in D:\a\_work\1\s\External\debugger-libs\Mono.Debugging.Soft\SoftDebuggerBacktrace.cs:line 128
   at Mono.Debugging.Client.Backtrace.GetFrame(Int32 index, Int32 fetchMultipleCount) in D:\a\_work\1\s\External\debugger-libs\Mono.Debugging\Mono.Debugging.Client\Backtrace.cs:line 61
   at Mono.Debugging.VisualStudio.ExceptionsAdapter.GetExceptionInfo(Backtrace backTrace, EvaluationOptions options, Boolean isUnhandled) in D:\a\_work\1\s\src\Debugging.VisualStudio.Vsix\Integration\ExceptionsAdapter.cs:line 96
   at Mono.Debugging.VisualStudio.ExceptionsAdapter.OnExceptionThrown(Object sender, TargetEventArgs args) in D:\a\_work\1\s\src\Debugging.VisualStudio.Vsix\Integration\ExceptionsAdapter.cs:line 56
   at Mono.Debugging.Client.DebuggerSession.OnTargetEvent(TargetEventArgs args) in D:\a\_work\1\s\External\debugger-libs\Mono.Debugging\Mono.Debugging.Client\DebuggerSession.cs:line 1254
   at Mono.Debugging.Soft.SoftDebuggerSession.HandleBreakEventSet(Event[] es, Boolean dequeuing) in D:\a\_work\1\s\External\debugger-libs\Mono.Debugging.Soft\SoftDebuggerSession.cs:line 2244
   at Mono.Debugging.Soft.SoftDebuggerSession.EventHandler() in D:\a\_work\1\s\External\debugger-libs\Mono.Debugging.Soft\SoftDebuggerSession.cs:line 1802
```

The app runs fine without exception, but when I try to throw the exception and catch it, it looks like Debugger is unable to load PDB and get the stack frame information, due to this, actaul exception information is no longer available.

I have tried to change Debug Symbol formats to full/embedded/portable... nothing works.

### Steps to Reproduce

1. Update Visual Studio 2022 to latest version.
2. Create a simple app, throw an exception when you click a button or something.

### Did you find any workaround?

1. Deleting the bin/obj folders fixes this issue.

**Comment by ackava** [comment]
I was able to fix this issue by manually deleting bin/obj folders in each project in solution.

**Comment by rockytauhid** [comment]
I have tried the following unsuccessfully
- deleting the bin and obj folders
- delete all and clone repository
- revert update to the previous version that's working (17.6.4 - 17.6.3)

The one that is working is going back to 17.4

**Comment by arvindrajachourasiya** [comment]
Same issue I am facing. `System.BadImageFormatException: Read out of bounds` sometimes it comes on app starting, smt's after app starting, smt's after one or two pages browsing. Its random. but it comes every time I run the app.

**Comment by sesitosvdx** [comment]
I am reproducing the same. The only workaround I found was to change a ListView to a CollectionView.

**Comment by mosammo** [comment]
Seems fixed in VS 17.7.7 Preview 6

**Comment by jpobst** [comment]
This seems to have been tracked by the IDE team here:
https://developercommunity.visualstudio.com/t/Debugging-XamarinForms-Android-app-throw/10392893

Should be fixed in VS 17.7.
